src/Agent/NewRelic/Agent/Core/DataTransport/ConnectionManager.cs

```diff
@@ -170,15 +175,7 @@ private void HandleConnectionException(Exception ex)
 
         private void Disconnect()
         {
-            _lockSemaphore.Wait();
-            try
-            {
-                _connectionHandler.Disconnect();
-            }
-            finally
-            {
-                _lockSemaphore.Release();
-            }
+            _connectionHandler.Disconnect();
```

Comment:
Why does this not need to be guarded anymore?

Reply:
Turns out you can't call `_lockSemaphore.Wait()` from one method then call it again from a nested method call. `Disconnect()` is being called from `Reconnect()`, which already waits for the semaphore. (`Disconnect()` is also being called from `OnCleanShutdown()` which isn't locked, but I'll throw that in real quick.)

Reply:
That's a good catch. Yeah, with this change, and the guarded region being particular large makes it harder to track the code that is already guarded, and which code isn't.